src/ocnicepost.fd/outputvars.F90

```diff
@@ -0,0 +1,146 @@
+module outputvars
+
+  implicit none
+
+  integer, parameter :: maxvars = 40          !< The maximum number of variables written to a file
+
+  type :: vardefs
+     character(len= 10)   :: input_var_name    !< A variable's  input variable name
+     character(len= 10)   :: output_var_name   !< A variable's output variable name
+     character(len=120)   :: long_name         !< A variable's long name
+     character(len= 20)   :: units             !< A variable's unit
+     character(len= 10)   :: var_remapmethod   !< A variable's mapping method
+     integer              :: var_dimen         !< A variable's dimensionality
+     character(len=  4)   :: var_grid          !< A variable's input grid location; all output locations are on cell centers
+     character(len= 10)   :: var_pair          !< A variable's pair
+     character(len=  4)   :: var_pair_grid     !< A pair variable grid
+     real                 :: var_fillvalue     !< A variable's fillvalue
+  end type vardefs
+
+  type(vardefs), public :: outvars(maxvars)    !< Attribute definitions for the variables
+
+contains
+
+  subroutine ocnvars_typedefine
+
+    ! local variables
+    integer :: ii = 0
+
+    !set defaults
+    outvars(:)%input_var_name=''
+    outvars(:)%var_grid  = 'Ct'
+    outvars(:)%var_remapmethod  = 'bilinear'
+    outvars(:)%var_dimen = 2
+    outvars(:)%var_pair = ''
+    outvars(:)%var_pair_grid = ''
+    outvars(:)%long_name = ''          ! obtained from input file
+    outvars(:)%units = ''              ! obtained from input file
+    outvars(:)%var_fillvalue = -1.0    ! obtained from input file
+
+    ! 2D states with native grid location on cell centers; remapped bilinearly
+    ii = ii + 1; outvars(ii)%input_var_name  = 'SSH'
+    ii = ii + 1; outvars(ii)%input_var_name  = 'SST'
+    ii = ii + 1; outvars(ii)%input_var_name  = 'SSS'
+    ii = ii + 1; outvars(ii)%input_var_name  = 'speed'
+    !ii = ii + 1; outvars(ii)%input_var_name  = 'mld'
+    ii = ii + 1; outvars(ii)%input_var_name  = 'ePBL'
+    ii = ii + 1; outvars(ii)%input_var_name  = 'MLD_003'
+    ii = ii + 1; outvars(ii)%input_var_name  = 'MLD_0125'
```

Comment:
@DeniseWorthen You are hard coding variables below (see above). Hard coding paths, variables, etc., that can be read via an external file is not permitted.